File: ProjectEulerPython/problems/problem_006.py
from problems.problem import Problem

def sum_squares_1_to_n(n: int) -> int:
  """For proof of this formula, see https://math.stackexchange.com/questions/48080/sum-of-first-n-squares-equals-fracnn12n16"""
  return n * (n + 1) * (2 * n + 1) // 6

def sum_1_to_n_squared(n: int) -> int:
  """For proof of the formula (first line), see https://en.wikipedia.org/wiki/Triangular_number"""
  sum = n * (n + 1) // 2
  return sum * sum

class Problem006(Problem):

  def calculate_answer(self) -> int:
    n = 100
    sum_squares = sum_squares_1_to_n(n)
    sum_squared = sum_1_to_n_squared(n)

    self.print_detail("Sum of squares 1 to " + str(n) + ":")
    self.print_detail(sum_squares)
    self.print_detail("Sum 1 to " + str(n) + " squared:")
    self.print_detail(sum_squared)

    return sum_squared - sum_squares

File: ProjectEuler/Problems/Problem006.cs
using Microsoft.Extensions.Logging;

namespace ProjectEuler.Problems
{
    public class Problem006 : Problem<long>
    {
        public Problem006(ILogger<Problem006> logger) : base(logger) { }

        protected override long CalculateAnswer()
        {
            var n = 100;
            var sumOfSquares1ToN = SumOfSquares1ToN(n);
            var sum1ToNSquared = Sum1ToNSquared(n);

            Logger.LogDebug("Sum Of Squares 1 To {N}:", n);
            Logger.LogDebug("{SumOfSquares1ToN}", sumOfSquares1ToN);
            Logger.LogDebug("Sum 1 To {N} Squared:", n);
            Logger.LogDebug("{Sum1ToNSquared}", sum1ToNSquared);

            return sum1ToNSquared - sumOfSquares1ToN;
        }

        private static long SumOfSquares1ToN(long n)
        {
            return n * (n + 1L) * (2 * n + 1L) / 6L;
        }

        private static long Sum1ToNSquared(long n)
        {
            var sum = n * (n + 1L) / 2L;
            return sum * sum;
        }
    }
}
